### PR TITLE
added variable pixel camera sizes

### PR DESCRIPTION
**Pull request by nickwalton**

### CODE REVIEW

**Comment by nickwalton** [comment]
Adds ability for the user to change the pixel camera size. 

Note: This will require a new config for the packaged worlds including the "window_height","window_width","camera_height", and "camera_width" parameters instead of "resx" and "resy"

Also: Larger resolutions of the pixel camera can dramatically decrease performance. If you want to use larger resolutions it's recommended to use a primary player camera.

**Comment by nickwalton** [comment]
Adds the ability for the user to select variable pixel camera sizes. Note: The Json needs to be updated with window_height, window_width, camera_height, and camera_widht

**Comment by nickwalton** [comment]
Also is this codacy check worthwhile to fix?

File: holodeck/environments.py [hunk 1]
```diff
@@ -288,10 +292,12 @@ def __linux_start_process__(self, binary_path, task_key, gl_version, verbose):
...
                                                 initial_value=0)
         self._world_process = subprocess.Popen([binary_path, task_key, '-HolodeckOn', '-opengl' + str(gl_version),
-                                                '-LOG=HolodeckLog.txt', '-ResX=' + str(self._width),
-                                                '-ResY=' + str(self._height), '--HolodeckUUID=' + self._uuid],
-                                               stdout=out_stream,
-                                               stderr=out_stream)
+                                                '-SILENT', '-LOG=HolodeckLog.txt', '-ResX=' + str(self._window_width),
+                                                "-ResY=" + str(self._window_height), '-CamResX=' + str(self._camera_width),
+                                                "-CamResY=" + str(self._camera_height), "--HolodeckUUID=" + self._uuid],
+                                               stdout=open(os.devnull, 'w'),
```
Comment (joshgreaves) [inline holodeck/environments.py]: These lines should be:
```
stdout=out_stream
stderr=out_stream
```

File: holodeck/environments.py [hunk 2]
```diff
@@ -56,8 +56,9 @@ def __init__(self, agent_name, agent_type, sensors=None):
 class HolodeckEnvironment(object):
     """The high level interface for interacting with a Holodeck world"""
 
-    def __init__(self, agent_definitions, binary_path=None, task_key=None, height=512, width=512,
-                 start_world=True, uuid="", gl_version=4, verbose=False):
+    def __init__(self, agent_definitions, window_height=512, window_width=512, camera_height=256, camera_width=256,
+                 binary_path=None, task_key=None,start_world=True, uuid="", gl_version=4):
```
Comment (joshgreaves) [inline holodeck/environments.py]: Nits: Should be a space between `task_key=None,start_world=True`. Quickly run reformat file in pycharm to make sure everything is formatted correctly

File: holodeck/environments.py [hunk 3]
```diff
@@ -303,8 +309,12 @@ def __windows_start_process__(self, binary_path, task_key, verbose):
         import win32event
         out_stream = sys.stdout if verbose else open(os.devnull, 'w')
         loading_semaphore = win32event.CreateSemaphore(None, 0, 1, "Global\\HOLODECK_LOADING_SEM" + self._uuid)
-        self._world_process = subprocess.Popen([binary_path, task_key, '-HolodeckOn', '-LOG=HolodeckLog.txt',
-                                                '-ResX=' + str(self._width), " -ResY=" + str(self._height),
+
+        self._world_process = subprocess.Popen([binary_path, task_key, '-HolodeckOn', '-SILENT', '-LOG=HolodeckLog.txt',
```
Comment (joshgreaves) [inline holodeck/environments.py]: Remove `-SILENT` flag

Comment (joshgreaves) [reply holodeck/environments.py]: And for the Linux subprocess code ^

File: holodeck/holodeck.py [hunk 4]
```diff
@@ -13,7 +13,8 @@ class GL_VERSION(object):
     OPENGL3 = 3
 
 
-def make(world_name, gl_version=GL_VERSION.OPENGL4, resolution=None, verbose=False):
```
Comment (joshgreaves) [inline holodeck/holodeck.py]: Looks like a lot of this has removed the verbose code. Make sure the flag still exists and works

File: holodeck/holodeck.py [hunk 5]
```diff
@@ -30,10 +31,13 @@ def make(world_name, gl_version=GL_VERSION.OPENGL4, resolution=None, verbose=Fal
     param_dict["start_world"] = True
     param_dict["uuid"] = str(uuid.uuid4())
     param_dict["gl_version"] = gl_version
-    param_dict["verbose"] = verbose
```
Comment (joshgreaves) [inline holodeck/holodeck.py]: And here

File: holodeck/environments.py (hunk 2 above)
Comment (joshgreaves) [reply holodeck/environments.py]: Also, verbose flag has been removed

**Comment by nickwalton** [comment]
Made a new pull request

**Pull request closed without merging**